index.js

```diff
@@ -15,20 +17,27 @@ module.exports.parse = (event) => {
                         .match(/name="[a-zA-Z_]+([a-zA-Z0-9_]*)"/)[0]
                         .split('=')[1]
                         .match(/[a-zA-Z_]+([a-zA-Z0-9_]*)/)[0]
-                    ] = {
-                    type: 'file',
-                    filename: item
-                        .match(/filename="[\w-\.\/]+"/)[0]
-                        .split('=')[1]
-                        .match(/[\w-\.\/]+/)[0],
-                    contentType: item
-                        .match(/Content-Type: .+\r\n\r\n/)[0]
-                        .replace(/Content-Type: /, '')
-                        .replace(/\r\n\r\n/, ''),
-                    content: Buffer.from(item
-                        .split(/\r\n\r\n/)[1]
-                        .replace(/(\r\n)+--/, ''), 'latin1'),
-                };
+                ] = {
+                        type: 'file',
+                        filename: item
+                            .match(/filename="[\w-\. \/]+"/)[0]
+                            .split('=')[1]
+                            .match(/[\w-\.]+/)[0],
+                        contentType: item
+                            .match(/Content-Type: .+\r\n\r\n/)[0]
+                            .replace(/Content-Type: /, '')
+                            .replace(/\r\n\r\n/, ''),
+                        content: (spotText && item
+                            .match(/Content-Type: .+\r\n\r\n/)[0]
+                            .replace(/Content-Type: /, '')
+                            .replace(/\r\n\r\n/, '')
+                            .match(/text/)
+                        ) ? item
+                            .split(/Content.+\r\n\r\n/)[1]
+                            .replace(/(\r\n)+(--)+/, '') : Buffer.from(item
+                                .split(/Content.+\r\n\r\n/)[1]
```

Comment:
Adding `Content.+` here prevents the split from matching multiple elements where `\r\n\r\n` could exist in the payload